pkgs/utils/src/mix.zig

```diff
@@ -0,0 +1,78 @@
+const std = @import("std");
+/// Constant which represents an empty structure field
+pub const Empty = std.builtin.Type.StructField{
+    .name = "",
+    .type = undefined,
+    .default_value_ptr = null,
+    .is_comptime = false,
+    .alignment = 0,
+};
+
+/// Ensures the type matches the wanted type kind
+inline fn ensure(comptime T: type, comptime kind: std.builtin.TypeId) ?std.meta.TagPayload(std.builtin.Type, kind) {
+    return if (@typeInfo(T) == kind) @field(@typeInfo(T), @tagName(kind)) else null;
+}
+
+fn indexByName(comptime fields: []const std.builtin.Type.StructField, name: []const u8) ?usize {
+    for (fields, 0..) |field, i| {
+        if (std.mem.eql(u8, field.name, name)) return i;
+    }
+    return null;
+}
+
+/// Mixes fields from structure extend into structure super
+pub fn Mix(comptime Super: type, comptime Extend: type) type {
+    const superInfo = ensure(Super, .@"struct") orelse @panic("Super type must be a struct");
+    const extendInfo = ensure(Extend, .@"struct") orelse @panic("Extend type must be a struct");
```

Comment:
not that I recommend doing that in this PR, but none of them need to be a struct, you could just `switch` on the type. Let's not complicate this PR though.

Reply:
most of this is copied snipped so yea we can trim it down later as you said